sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -96,11 +96,6 @@ case class CreateViewCommand(
   }
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
-    // If the plan cannot be analyzed, throw an exception and don't proceed.
-    val qe = sparkSession.sessionState.executePlan(child)
-    qe.assertAnalyzed()
-    val analyzedPlan = qe.analyzed
```

Comment:
Actually this is a regression. `CreateViewCommand` can be created from `CreateViewStatement` in `ResolveSessionCatalog`. However, `ResolveSessionCatalog` is inside `extendedResolutionRules`, which means the `analyzedPlan` doesn't go through `postHocResolutionRules`.

I think we need to make `CreateViewCommand.analyzedPlan` a real child, so that it can go through all the analyzer rules.

Reply:
Good catch. One issue with "make CreateViewCommand.analyzedPlan a real child" is that it will become an optimized plan, which will affect the caching. I can try to make certain command's children to skip optimizer. WDYT?

Reply:
It will be great to have such a mechanism to skip optimizer, useful to `CacheTableAsSelect` as well.

Reply:
OK, I will give it a shot.

Reply:
@cloud-fan I created https://github.com/apache/spark/pull/32032 to handle this scenario. Please let me know what you think. TIA!